pydy/codegen/ode_function_generators.py

```diff
@@ -417,7 +430,7 @@ def _convert_specifieds_dict_to_array(self, x, t, r):
             # TODO : Not sure if this is the best check here.
             if isinstance(type(k), UndefinedFunction):
                 k = (k,)
-            idx = [self.specifieds.index(symmy) for symmy in k]
+            idx = [list(self.specifieds).index(symmy) for symmy in k]
```

Comment:
Actually, I don't see where you changed this to a set? It looks like you didn't. Did you?